mg/cmd_test.go

```diff
@@ -0,0 +1,189 @@
+package mg_test
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"testing"
+
+	"margo.sh/mg"
+)
+
+func TestCmdOutputWriter_Copy(t *testing.T) {
+	var called1, called2 bool
+	u1 := func(*mg.CmdOutputWriter) { called1 = true }
+	u2 := func(*mg.CmdOutputWriter) { called2 = true }
+
+	w := &mg.CmdOutputWriter{}
+	w.Copy(u1, u2)
+	if !called1 {
+		t.Error("u1 wasn't called")
+	}
+	if !called2 {
+		t.Error("u2 wasn't called")
+	}
+}
+
+var errNotEnoughSize = fmt.Errorf("not enough size")
+
+type limitWriter []byte
+
+func (l *limitWriter) Write(p []byte) (n int, err error) {
```

Comment:
Since you don't intend to overwrite the pointer, just use the (slice) value directly `func (l limitWriter)`